src/core/ChainRequest.js

```diff
@@ -0,0 +1,56 @@
+import fetch from 'node-fetch';
+
+const api_port = '4000';
+const url = 'http://localhost:';
+
+const request = (route, method, port = api_port, body = {}) => {
+  return new Promise(async (resolve, reject) => {
```

Comment:
you don't need a wrapper promise here--just make `request` itself an async function, and throw an error if `status` is not a good value. note additionally that `!status` will still be false even if the server returns an error code like HTTP 500--perhaps you want to use `result.ok` instead (see docs [here](https://github.com/bitinn/node-fetch#handling-client-and-server-errors))

Reply:
The `status` value itself  is what causes some issues here. By convention I always thought that a returned success given by an integer would be `1` but in the case of these requests to the `ex_testchain` a successful request has a status of `0`.

Seeing that doc, the backend clearly isn't responding with the correct err/response codes.

Reply:
ack, never mind my comment about `status`. i misread the code here and mixed up the HTTP status code with the `status` property within the JSON response. two different things. sorry for the confusion